### PR TITLE
docs(start/overview): add `Search Params` section

### PR DESCRIPTION
**Pull request by OlegDev1**
Added a Search Params section to the Feature Overview in the documentation. It includes a simple description with references, along with an example showing the main concepts.

### CODE REVIEW

**Comment by brookslybrand** [comment]
Thanks @OlegDev1 for noticing and filling out this section.

You can also change the search param with `useNavigate` and `Link`. It might be good to show some examples there, so we don't accidentally encourage people to reach for `setSearchParams` in every situation.

You could maybe just add comments to the alternative APIs within your code example


```jsx
return (
  <>
    <p>The value of the search param with key q is: {query}</p>
    <button onClick={handleClick}>Change search params</button>
    {/* <Link to={{ search: 'q=example' }}>Change search params</Link> */}
  </>
);
```

**Comment by OlegDev1** [comment]
> Thanks @OlegDev1 for noticing and filling out this section.
> 
> You can also change the search param with `useNavigate` and `Link`. It might be good to show some examples there, so we don't accidentally encourage people to reach for `setSearchParams` in every situation.
> 
> You could maybe just add comments to the alternative APIs within your code example
> 
> ```js
> return (
>   <>
>     <p>The value of the search param with key q is: {query}</p>
>     <button onClick={handleClick}>Change search params</button>
>     {/* <Link to={{ search: 'q=example' }}>Change search params</Link> */}
>   </>
> );
> ```

Sure! I have added another example using useNavigate and Link

File: docs/start/overview.md [hunk 1]
```diff
@@ -697,7 +697,57 @@ When you get better at React Router, you get better at the web platform.
 
 ## Search Params
 
-<docs-info>TODO:</docs-info>
+React Router can read and modify the [query string][querystring]. This process is handeled with [`URLSearchParams`][urlsearchparams]. We often get and set the Search Params using `useSearchParams`.
```
Comment (brookslybrand) [inline docs/start/overview.md:700]:
```suggestion
React Router can read and modify the [query string][querystring]. You can set the search params using `useSearchParams`, which returns [`URLSearchParams`][urlsearchparams] and a setter function.
```

File: docs/start/overview.md [hunk 2]
```diff
@@ -697,7 +697,57 @@ When you get better at React Router, you get better at the web platform.
...
+  return (
+    <>
+      <p>The value of the search param with key q is: {query}</p>
+      <button onClick={handleClick}>Change search params</button>
+    </>
+  );
+}
+```
+
+You can also use `Link` or `useNavigate` to change the Search Params.
```
Comment (brookslybrand) [inline docs/start/overview.md:722]:
```suggestion
You can also use `Link` or `useNavigate` to change the search params.
```

**Review by brookslybrand — APPROVED** [review]
Looks good, thanks @OlegDev1!

**Pull request merged**